TodoListSPA/app.js

```diff
@@ -0,0 +1,238 @@
+'use strict';
+
+config.callback = loggedin;
+var authenticationContext = new AuthenticationContext(config);
+
+if (!config.popUp) {
+    if (authenticationContext.isCallback(window.location.hash)) {
+        authenticationContext.handleWindowCallback();
+    }
+    else {
+        var user = authenticationContext.getCachedUser();
+        if (user && window.parent === window && !window.opener) {
+            // Display the user
+            displayUserAndShowSignOutButton(user);
+
+            // Call the protected API to show the content of the todo list
+            acquireAnAccessTokenAndCallTheProtectedService();
+        }
+    }
+}
+
+
+/**
+ * Entry point: called when the user clicks on the "Display the todo list" button
+ */
+function displayTodoList() {
+    var user = authenticationContext.getCachedUser();
+    if (user) {
+        onLogin(null, user);
+    }
+    else {
+        showProgress("Getting user's identity");
+        authenticationContext.login();
+    }
+}
+
+/**
+ * Entry point: called when the user clicks on the "Logout" button
+ */
+function signOut() {
+    authenticationContext.logOut();
+}
+
+/**
+ * Callback, triggered after authentication.login() got called and the login interaction
+ * with the user happened.
+ * @param {string} errorDescription - Description of the error (as returned from the STS)
+ * @param {string} idToken - Id token in case of success
+ * @param {string} error - Error code
+ */
+function loggedin(errorDescription, idToken, error) {
```

Comment:
What's the point in passing the ID Token into this function? Don't necessarily need to use it for the sample, but I don't know how it would be used.